application/config/config_test.go

```diff
@@ -161,43 +161,62 @@ func Test_loadFile(t *testing.T) {
 }
 
 func TestSnykCodeApi(t *testing.T) {
+	t.Setenv("DEEPROXY_API_URL", "")
 	t.Run("endpoint not provided", func(t *testing.T) {
 
-		codeApiEndpoint, _ := getCodeApiUrlFromCustomEndpoint("")
+		codeApiEndpoint, _ := getCodeApiUrlFromCustomEndpoint("", false)
 		assert.Equal(t, "https://deeproxy.snyk.io", codeApiEndpoint)
+
+		codeApiEndpoint, _ = getCodeApiUrlFromCustomEndpoint("", true)
```

Comment:
This and test cases below can be simplified using table-driven tests. Using an array of test inputs and expected outputs for testing is more readable and maintainable.